projects/HotFire/config/UOStateMachine.cpp

```diff
@@ -35,6 +36,24 @@ void UOStateMachine::StartFillingEXT()
 		END_TRANSITION_MAP(NULL)
 }
 
+// Abort external event
+void UOStateMachine::AbortEXT()
+{
+	BEGIN_TRANSITION_MAP					   // - Current State -
+		TRANSITION_MAP_ENTRY(EVENT_IGNORED)	   // ST_INIT
+		TRANSITION_MAP_ENTRY(EVENT_IGNORED)	   // ST_WAIT_FOR_INIT
+		TRANSITION_MAP_ENTRY(ST_ABORT_FILLING) // ST_WAIT_FOR_FILLING
+		TRANSITION_MAP_ENTRY(ST_ABORT_FILLING) // ST_FILLING
+		TRANSITION_MAP_ENTRY(ST_ABORT_FILLING) // ST_WAIT_FOR_IGNITION
+		TRANSITION_MAP_ENTRY(ST_ABORT_BURN)	   // ST_IGNITION
+		TRANSITION_MAP_ENTRY(ST_ABORT_BURN)	   // ST_FULL_BURN
+		TRANSITION_MAP_ENTRY(EVENT_IGNORED)	   // ST_FINAL_VENTING
```

Comment:
Do we want an abort for `ST_FINAL_VENTING`

Reply:
What state would that transition to?

Reply:
Not sure. I guess this state is already an "abort" state, just a successful one, so it wouldn't make sense to abort from that.